model/event/DeliveryCreatedEvent.php

```diff
@@ -19,16 +19,24 @@
  */
 namespace oat\taoDeliveryRdf\model\event;
 
+use core_kernel_classes_Resource;
 use oat\tao\model\webhooks\WebhookSerializableEventInterface;
 
 class DeliveryCreatedEvent extends AbstractDeliveryEvent implements WebhookSerializableEventInterface
 {
+    private $testUri;
+
     /**
-     * @param String $deliveryUri
+     * @param string $deliveryUri
+     * @param string $testUri
+     *
+     * @throws \core_kernel_persistence_Exception
      */
-    public function __construct($deliveryUri)
+    public function __construct(core_kernel_classes_Resource $delivery)
     {
-        $this->deliveryUri = $deliveryUri;
+        $testProperty = new \core_kernel_classes_Property('http://www.tao.lu/Ontologies/TAODelivery.rdf#AssembledDeliveryOrigin');
+        $this->deliveryUri = $delivery->getUri();
+        $this->testUri = $delivery->getOnePropertyValue($testProperty)->getUri();
```

Comment:
I discourage to start process from constructor, e.g. access to database. Also it is possible to have a reference as delivery origin to a deleted test, could u create a test to ensure this point please